packages/daemon/src/daemon.js

```diff
@@ -227,6 +227,29 @@ const makeEndoBootstrap = (
     return { external, internal: undefined };
   };
 
+  /**
+   * Creates a controller for a `lookup` formula. The external facet is the
+   * resolved value of the lookup.
+   *
+   * @param {string} agentFormulaIdentifier
+   * @param {string[]} path
+   * @param {import('./types.js').Terminator} terminator
+   */
+  const makeControllerForLookup = async (
+    agentFormulaIdentifier,
+    path,
+    terminator,
+  ) => {
+    terminator.thisDiesIfThatDies(agentFormulaIdentifier);
```

Comment:
In the future, we should also have a watcher on the store such that any change in the names along the lookup path causes this formula to be terminated.

Reply:
Now: Add terminate hook that does a reverse lookup of the resolved `lookup` formula value and calls `terminate()` on it. Return early if the reverse lookup returns an empty list.

Reply:
Ref: #2041